mindmeld/models/query_features.py

```diff
@@ -537,13 +537,25 @@ def _extractor(query, resources):
                                 entity.entity.type, entity.entity.value.get("grain"), j
                             )
                         )
-                        feat_seq[i - j][feat_name] = 1
+                        if feat_name in feat_seq[i - j]:
+                            feat_seq[i - j][feat_name] += 1
+                        else:
+                            feat_seq[i - j][feat_name] = 1
+
                         feat_name = "sys_candidate|type:{}|granularity:{}|pos:{}|log_len".format(
                             entity.entity.type, entity.entity.value.get("grain"), j
                         )
-                        feat_seq[i - j][feat_name] = math.log(
-                            len(entity.normalized_text)
-                        )
+                        if feat_name in feat_seq[i - j]:
+                            feat_seq[i - j][feat_name].append(len(entity.normalized_text))
+                        else:
+                            feat_seq[i - j][feat_name] = [len(entity.normalized_text)]
+
+        for token_features in feat_seq:
+            for feature, value in token_features.items():
+                if feature.endswith('log_len'):
+                    token_features[feature] = math.log(float(sum(value)) / len(value))
+                else:
+                    token_features[feature] = math.log(value)
```

Comment:
Haven't really thought through how this would would affect things, but since we use a feature scaler (defaults to [MaxAbsScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MaxAbsScaler.html#sklearn.preprocessing.MaxAbsScaler)) anyway, maybe we can use the raw counts?

Reply:
Just wanted to note that for the features ending in `log_len` we were already taking the log.

Reply:
@vembar Good point, the maxabs is a default option for the entity rec features. I will try with the raw counts

Reply:
@jjjacksn Maybe I am misunderstanding, but in this line of code, we have taken the log of the expression exactly once, ie the log for this feature has not been taken before this line of code.

Reply:
MaxAbsScaler does linear scaling onto the interval, whereas the `log(length)` is just meant to balance small differences vs large differences, e.g. the difference between `1` and  `100` scales the odds more like the difference between `100` and `100000000` (three times as long) than the difference between `100` and `10000` (2 chars longer).

Reply:
@serapio Thanks for that comment. I think it makes sense to continue using log_len, but for the other feature, the log count , where the count refers to the total number of system entities that overlap with a token, I think the log scaling doesnt help much since the count distribution looks pretty linear.

I did an analysis on the hr_blueprint and banking_assistant between master vs log vs no-log here: https://cisco-my.sharepoint.com/:x:/p/vijayrk/EQJzxhsBPWhHipofNShGEZwBv9rAXW9GlywB4gqTzEvlBQ?e=uBSmke. I don't think there are any clear patterns in the data ie not much difference between the three systems.

Reply:
I am fine with either log or no log for the counts. The only issue if the developer doesnt use maxabs, they might be in trouble, since the log operation is in itself a type of scaler function. Thoughts? @vembar @serapio

Reply:
Looking at the results, it does seem like the results are varying slightly based on log or no_log for numeric entities, but yeah, there's no clear pattern. It's also a good point that the developer may choose remove the scaler. So all things considered, makes sense to keep the log operation.